lib/engine/step/g_1856/dividend.rb

```diff
@@ -24,7 +24,7 @@ def process_dividend(action)
           revenue = @game.routes_revenue(routes) - penalty
           kind = action.kind.to_sym
           payout = dividend_options(entity)[kind]
-
+          @game.process_derived_action(Engine::Action::Bonus.new(entity)) unless @game.loading
```

Comment:
So I have a different way of thinking about this.

For instance in the 1867 token case.
The step should occur as normal, and the method should return a

def automatic_action(entity)
  Engine::Action::Pass.new(entity) if rather_expensive_check_returns_that_only_option_is_pass
end


This would return either nil for no action, or an action to be added.

process_action when it's finished it's normal actions would check the current step to see if it's got a derived action and if so process it until it runs out of derived actions. These derived actions would be passed upwards to the callee so that they can be added to the json/database.

Reply:
OK so you are suggesting is that instead of calling `@game.process_derived_action(<new action>)` inside `process_#{action}` to have `process_#{action}` return an array of actions, or to have `automatic_action(entity)` be a separate method that gets invoked by the code that called `process_#{action}`?

Reply:
At the end of process_action (or an alternative say generate_derived_actions which can get called when trying to write actions) it'll keep trying to see if there's an automatic action available to it.

This could be limited to the UI and submitted as a bunch of actions.